website/server/libs/chatReporting/inboxChatReporter.js

```diff
@@ -0,0 +1,106 @@
+import nconf from 'nconf';
+import { model as User } from '../../models/user';
+
+import ChatReporter from './chatReporter';
+import {
+  NotFound,
+} from '../errors';
+import { getGroupUrl, sendTxn } from '../email';
+import slack from '../slack';
+import apiError from '../apiError';
+
+import _find from 'lodash/find';
+import * as inboxLib from '../inbox';
+
+const FLAG_REPORT_EMAILS = nconf.get('FLAG_REPORT_EMAIL').split(',').map((email) => {
+  return { email, canSend: true };
+});
+
+export default class InboxChatReporter extends ChatReporter {
+  constructor (req, res) {
+    super(req, res);
+
+    this.user = res.locals.user;
+    this.inboxUser = res.locals.user;
+  }
+
+  async validate () {
+    this.req.checkParams('messageId', apiError('messageIdRequired')).notEmpty();
+
+    let validationErrors = this.req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    if (this.user.contributor.admin && this.req.query.userId) {
+      this.inboxUser = await User.findOne({_id: this.req.query.userId});
+    }
+
+    const messages = await inboxLib.getUserInbox(this.inboxUser);
```

Comment:
`inboxUser` is set at line [24](https://github.com/HabitRPG/habitica/pull/10420/files/2c250bfcd9820f95e74f45c817f17d869c1cbbe9#diff-609388719783e1ac10548c4df5e90399R24) 

but I guess this condition (line 33) isn't needed anymore, since there won't be a frag-o-rama link to delete it by an admin/mod, I'll remove this ^^